torchserve/datadog_checks/torchserve/inference_api_check.py

```diff
@@ -0,0 +1,33 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urljoin
+
+from datadog_checks.base import AgentCheck
+
+
+class TorchserveInferenceAPICheck(AgentCheck):
+    __NAMESPACE__ = 'torchserve'
+    SERVICE_CHECK_NAME = 'inference_api.health'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.tags = []
+        self.check_initializations.append(self.parse_config)
+
+    def parse_config(self):
+        self.tags = self.instance.get("tags", [])
+
+    def check(self, _):
+        ping_url = urljoin(self.instance['inference_api_url'], 'ping')
+
+        try:
+            self.log.debug("Querying URL: [%s]", ping_url)
+            response = self.http.get(ping_url)
+            self.log.debug("Inference API `response`: [%s]", response)
+            response.raise_for_status()
+        except Exception:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self.tags)
```

Comment:
Should `tags` also contain a tag for the `ping_url`?

Reply:
Since we have `openmetrics_endpoint` for openmetrics, we can add `inference_api_url` and `management_api_url`. What do you think?

Reply:
added https://github.com/DataDog/integrations-core/pull/14608/commits/133d2fb218bf7bf139f3d8f7f9a2ee243bba0ddb